polaris-react/src/components/Select/Select.scss

```diff
@@ -43,13 +43,6 @@
     fill: var(--p-color-icon-disabled);
   }
 
-  #{$se23} & {
-    // stylelint-disable-next-line -- se23 specificity
-    .Icon svg {
-      fill: var(--p-color-icon-disabled);
-    }
-  }
```

Comment:
[This PR](https://5d559397bae39100201eedc1-tehbqainqm.chromatic.com/?path=/story/all-components-select--disabled)
[Production](https://storybook.polaris.shopify.com/?path=/story/all-components-select--disabled&globals=polarisSummerEditions2023:true)

<img width="1280" alt="Screenshot 2023-08-25 at 1 51 20 PM" src="https://github.com/Shopify/polaris/assets/20652326/fd7637fc-1906-4ea8-9ae6-49e6da9de2bc">